ReactSkia/components/RSkComponentScrollView.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 1994-2021 OpenTV, Inc. and Nagravision S.A.
+ *
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#pragma once
+
+#include "react/renderer/components/scrollview/ScrollViewShadowNode.h"
+
+#include "ReactSkia/components/RSkComponent.h"
+#include "ReactSkia/RSkSurfaceWindow.h"
+
+namespace facebook {
+namespace react {
+
+class RSkComponentScrollView final : public RSkComponent {
+ public:
+  RSkComponentScrollView(const ShadowView &shadowView , RSkSurfaceWindow * surface);
+
+  void mountChildComponent(
+    std::shared_ptr<RSkComponent> newChildComponent,
+    const int index) override;
+
+  void unmountChildComponent(
+    std::shared_ptr<RSkComponent> oldChildComponent,
+    const int index) override;
+
+  void updateComponentProps(
+    const ShadowView &newShadowView,
+    bool forceUpadate) override;
+
+  bool needEvent() override;
+
+  bool onHandleKey(
+    rnsKey eventKeyType) override;
+
+
+ protected:
+  void OnPaint(SkCanvas *canvas) override;
+
+ private:
+  sk_sp<SkPicture> getPicture();
+
+  bool isHorizontalScroll();
```

Comment:
Does RN has component with both Horizontal and Vertical scroll behavior enabled ? If no then fine.

Reply:
No. Only one behavior can be enabled at a time. Either horizontal or vertical.